### PR TITLE
Bump Go to 1.22

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
**Motivation:**
- Go 1.21 is reached EOL
- Migration to kubernetes Gateway API requires Go 1.22

**Changes:**
- Bumps Go to 1.22
- Kubernetes generated client for both v1alpha1 and v1beta1API version will be available inside `client` directory. Removed `clientv1alpha1` directory. Previously the client for v1alpha1 is generated in [separate directory due to the limitation of client-gen script](https://github.com/kserve/kserve/blob/e879a397b38fdcbebd8ec004c4da87916fb30a03/hack/update-codegen.sh#L34-L36).
- The generated OpenAPI will be available under `pkg/openapi` directory. Previously it was under `pkg/apis/serving/v1beta1`. Since the OpenAPI spec contains both v1alpha1 and v1beta1 specifications, putting it under `v1beta1` directory seems inappropriate.
- Addresses the breaking changes introduced by controller-tools
- Addresses the breaking changes introduced in client-gen, deepcopy-gen tools
- Bumps controller-gen version to 0.16.2
- Addresses breaking change introduced by knative queue pkg in `agent ` sidecar.
- Addresses a breaking change introduced by istio api.
- Bumps kubernetes api to 1.30
- Addresses the breaking changes introduced in OpenAPI package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Bump go version to 1.22
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,87 +1,94 @@
 module github.com/kserve/kserve
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.6
```
Comment (terrytangyuan) [inline go.mod]: Why do we need this?

File: go.mod [hunk 2]
```diff
@@ -1,87 +1,94 @@
 module github.com/kserve/kserve
 
-go 1.21
+go 1.22.0
```
Comment (terrytangyuan) [inline go.mod]: 1.22.5 should work and resolve the toolchain issue you likely had

Comment (spolti) [reply go.mod]: is it a requirement to bump it now?

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**